### PR TITLE
Unify `ConstantValueOpt` and `ConstantValue` properties

### PR DESCRIPTION
**Pull request by alrz**
There's always two members showing up for expressions: ConstantValue and ConstantValueOpt. This merges the two.

Note: instead of rename (done in second commit) could add `ConstantValue => ConstantValueOpt;` to reduce the diff size, but that would defeat the purpose.

### CODE REVIEW

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Comment by 333fred** [comment]
Build failures look legitimate

**Review by 333fred — APPROVED** [review]

**Comment by 333fred** [comment]
Thanks @alrz!

File: src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml [hunk 1]
```diff
@@ -2187,15 +2187,15 @@
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
     <Field Name="Argument" Type="BoundExpression" Null="disallow"/>
-    <Field Name="ConstantValueOpt" Type="ConstantValue" Null="disallow"/>
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:2190]: >Null="disallow" [](http://example.com/codeflow?start=56&length=15)

Why did we drop ```Null="disallow"``` part?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:2190]: @alrz Could you comment on this?

Comment (alrz) [reply src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml:2190]: Nullability is already inferred by the type itself. Annotations are required anyways but usages of `"disallow"` for reference types is not consistent across the file.

**Pull request merged**